src/ui/pages/Menu/components/ConnectWallet/ConnectWallet.tsx

```diff
@@ -116,21 +118,21 @@ const ConnectWallet = forwardRef<ConnectWalletOptionRef, object>(
       handleOpenVerify();
     };
 
-    const handleDeleteConnection = (data: ConnectionData) => {
+    const handleDeleteConnection = async (data: ConnectionData) => {
       actionInfo.current = {
         type: ActionType.None,
       };
-
-      // TODO: Implement delete wallet connection logic
-
+      await Agent.agent.peerConnectionMetadataStorage.deletePeerConnectionMetadataRecord(
+        data.id
+      );
+      const connections =
+        await Agent.agent.peerConnectionMetadataStorage.getAllPeerConnectionMetadata();
```

Comment:
Here we can just remove the connection by ID from the Redux array - rather than fetching from the DB again which is more expensive